fix: default to dry-run mode with upgrade-script

rjan90 opened the pull request:
Default to dry-run mode with `upgrade-contract-calibnet.sh` script

rvagg reviewed (approved):
nice, while you're in here would you mind upgrading the chain ID based on the model in the other scripts, something like:

```sh
if [ -z "$CHAIN_ID" ]; then
  CHAIN_ID=$(cast chain-id --rpc-url "$RPC_URL")
  if [ -z "$CHAIN_ID" ]; then
    echo "Error: Failed to detect chain ID from RPC"
    exit 1
  fi
fi
```

then use that rather than a magic number, it makes it more portable too

rjan90 commented:
Refactored in: https://github.com/FilOzone/pdp/pull/200/commits/2150b92274601c4d512845d0efed68e2b868a1f2
- Made it network agnostic, and figure out the chain detection via `cast chain-id --rpc-url`
- Renamed from `upgrade-contract-calibnet.sh` to `upgrade-contract.sh` to reflect this.
